Explorer/Assets/DCL/CommunicationData/URLHelpers/URN.cs

```diff
@@ -10,19 +12,15 @@ public readonly struct URN
         private const int THIRD_PARTY_V2_SHORTEN_URN_PARTS = 7;
         private const string THIRD_PARTY_PART_ID = "collections-thirdparty";
 
-        private readonly string lowercaseUrn;
+        private const uint CACHE_MAX_SIZE = 1024;
+        private const int CACHE_CLEAR_AMOUNT = 64;
+        private static readonly ConcurrentDictionary<URN, URN> SHORTENED_URNS_CACHE = new ();
```

Comment:
The ConcurrentDictionary only ensures any operations are atomic, but the caching function can still have race conditions